scapy/layers/l2.py

```diff
@@ -564,7 +565,8 @@ class EAP_TLS(EAP):
         BitField('S', 0, 1),
         BitField('reserved', 0, 5),
         ConditionalField(IntField('tls_message_len', 0), lambda pkt: pkt.L == 1),
-        XStrLenField('tls_data', '', length_from=lambda pkt: pkt.len - 10 if pkt.L == 1 else pkt.len - 6)
+        XStrLenField('tls_data', '',
```

Comment:
Maybe I did not understand, but the test on pkt.len is necessary (it must not be `None` in order to perform the computation).